adapters/coap/src/main/java/org/eclipse/hono/adapter/coap/CoapContext.java

```diff
@@ -407,6 +450,13 @@ public void respondWithCode(final ResponseCode responseCode, final String descri
      * @return The response code from the response.
      */
     public ResponseCode respond(final Response response) {
+        // TODO: remove System.out.prinln
+        System.out.println("adding time option to response (v2) [" + isResponseTimestampEnabled + "]" );
+        if (isResponseTimestampEnabled) {
+            // TODO: figure out the "correct" Option ID here.
+            final Option timeOption = new Option(0xff3c, System.currentTimeMillis());
```

Comment:
Hey again @boaks and sorry for the delay.

I took a look at the DemoServer and I think I'm understanding the benefit of using the option registry.

However, that would require me to update the californium library version being used in hono and I'd have to figure out where to initialize the registries. I found a method in `AbstractVertxBasedCoapAdapter#preStartup()` that looks like a promising place to perform such an initialization but since I'm just starting to get into the code base (and I don't really have much experience with the reactive programming paradigm) I'd need some confirmation that I'm on the right track.

So before I start moving in this direction, I'd like to get a couple of things clarified from @sophokles73 

- Would it be okay to upgrade the Californium library to 3.8 as part of this PR?
- Does the `preStartup()` method sound like a good place to initialize the Californium `StandardOptionRegistry` like @boaks is doing in the DemoServer `main` method he references above?